planning/mv-params.json

```diff
@@ -0,0 +1,151 @@
+the user submits something that has this format:
+
+{
+    "includes": [
+	{
+	    "name": "common",
+	    "params": [
+		{ "arg": "foo", "vals": [ "val1", "val2" ], "role": "all" },
+		{ "arg": "bar", "vals": [ "val3", "val4" ], "role": "client" }
+	    ]
+	},
+	{
+	    "name": "scenario1",
+	    "params": [
+		{ "arg": "zorg", "vals": [ "val5", "val6" ], "role": "server" },
+		{ "arg": "alpha", "vals": [ "val7", "val8" ], "role": "infra" }
+	    ]
+	},
+	{
+	    "name": "scenario2",
+	    "params": [
+		{ "arg": "beta", "vals": [ "val9", "val10" ], "role": "all" }
+	    ]
+	}
+    ],
+    "sets": [
+	[
+	    { "include": "common" },
+	    { "include": "scenario1" },
+	    { "arg": "gamma", "vals": [ "val11" ], "role": "client" }
+	],
+	[
+	    { "include": "common" },
+	    { "include": "scenario2" },
+	    { "arg": "delta", "vals": [ "val12", "val13" ], "role": "infra" }
+	],
+	[
+	    { "include": "common" },
+	    { "arg": "epsilon", "vals": [ "val14" ] }
```

Comment:
if "role" is missing we think we want it to default to "client" -- this is because in rickshaw workloads that only have a single tier refer to that tier as the client.  by setting client to the default very simple workloads (ie. single tier) will not require the user to needlessly list the "role" all over the place

Reply:
> I figured I would document what I think Andrew and I have been thinking for the multiplex mv-params input json and the process by which it is transformed into bench-params.json. By doing it as a PR we can discuss various lines and whether we think they should change in all parts of the process.
> 
> This is not intended to be merged, this is just to spark a collaborative discussion. Since multiplex is not really being used right now we have the opportunity to change things if we think we should. One example of this is I already changed "common" to "includes" and "include" which just seem much more appropriate to me.

I think it's important to get these documented somewhere, maybe in the sample section of README.

Reply:
Absolutely.  I just meant we should hash out the details before writing up formal documentation.